File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.cpp
class Solution {
public:
    int chalkReplacer(vector<int>& chalk, int k) {
        long long total = 0;
        for(const auto &x : chalk) {
            total += x;
        }
        k %= total;

        for(int i = 0; i < chalk.size(); i++) {
            if(chalk[i] <= k) k -= chalk[i];
            else return i;
        }

        return 0;
    }
};

File: 2006-find-the-student-that-will-replace-the-chalk/2006-find-the-student-that-will-replace-the-chalk.go
func chalkReplacer(chalk []int, k int) int {
    var total int64

    for _, c := range chalk {
        total += int64(c)
    }
    k = int(int64(k) % total)

    for i, v := range chalk {
        if v > k {
            return i
        }
        k -= v
    }
    return 0
}